### DISCUSSION TITLE
Not sure if this is a Zustand question, or a Javascript question

### DISCUSSION THREAD

**Discussion by leone6** [pmndrs/zustand#877] [Q&A] [answered]
please feel free to let me know if i am using zustand incorrectly... i am building a large react app that has many components. in each component, i am accessing the unique state for that component like this:

const someStateVariable = useStore((state) => state.uniqueComponentName.someStateVariable);

my problem is when i try to access this state inside the store. i would like a function in the store to be able to do something like this:

const someStateVariable= get().uniqueComponentName.someStateVariable;

BUT... the problem is that uniqueComponentName is dependent on args passed to the function so it changes frequently. is there a way to efficient swap out the uniqueComponentName right after get()?

i have avoiding this issue while setting this state because i have been using ES6 computed property values like this:

set(() => ({
     [key]: {
     // state here
     },
}));

where [key] is a computed property value. but how can i do this with get() ?

thank you!

**Comment by dai-shi** [comment, marked as answer]
You probably mean `get()[key].someStateVariable`?

**Reply by leone6** [reply]
this is great. i had no idea this syntax worked. thank you!
